packages/@lightningjs/ui-components/src/components/Badge/Badge.d.ts

```diff
@@ -33,16 +33,77 @@ export type BadgeStyle = {
   textStyle: lng.textures.TextTexture.Settings;
 };
 
-export default class Badge extends Base {
-  title?: string;
+declare namespace Badge {
+  export interface TemplateSpec extends Base.TemplateSpec {
+    /**
+     * path to image or inline SVG XML
+     */
+    icon?: string;
+
+    /**
+     * width of icon
+     */
+    iconWidth?: number;
+
+    /**
+     * height of icon
+     */
+    iconHeight?: number;
+
+    /**
+     * side of the text where icon will appear on (`left` or `right`)
+     */
+    iconAlign?: 'left' | 'right';
+
+    /**
+     * Badge text
+     */
+    title?: string;
+  }
+}
+
+declare class Badge<
+  TemplateSpec extends Badge.TemplateSpec = Badge.TemplateSpec,
+  TypeConfig extends lng.Component.TypeConfig = lng.Component.TypeConfig
+> extends Base<TemplateSpec, TypeConfig> {
+  /**
+   * path to image or inline SVG XML
+   */
   icon?: string;
-  iconAlign?: string;
-  iconHeight?: number;
+
+  /**
+   * width of icon
+   */
   iconWidth?: number;
+
+  /**
+   * height of icon
+   */
+  iconHeight?: number;
+
+  /**
+   * side of the text where icon will appear on (`left` or `right`)
+   */
+  iconAlign?: 'left' | 'right';
+
+  /**
+   * Badge text
+   */
+  title?: string;
+
+  // Events
+  /**
+   * This event is triggered when the Badge component has completed loading its content, and the background has been adjusted to fit the size of the text input.
+   */
+  $loadedBadge(): void;
```

Comment:
this should actually be documented in the [SignalMap](https://github.com/rdkcentral/Lightning/blob/8378d0e69752f476abedecc85568a27dbb63cbb0/docs/TypeScript/Components/TypeConfigs.md#signal-maps)

examples can be found in the [Artwork](https://github.com/rdkcentral/Lightning-UI-Components/pull/304) and [FocusManager](https://github.com/rdkcentral/Lightning-UI-Components/pull/305) PRs as well